src/Debug/boneWeightShader.ts

```diff
@@ -0,0 +1,80 @@
+import { Nullable } from '../types';
+import { Skeleton, Bone } from '../Bones';
+import { AbstractMesh } from '../Meshes';
+import { DynamicTexture } from '../Materials';
+import { Scene } from '../scene';
+import { Color3 } from '../Maths';
+
+
+interface IBoneWeightOptions{
+    name: Nullable<string>
+    skeleton: Skeleton
+    mesh: AbstractMesh    
+    renderingMode: number
+
+    //0-1 number, then a color3 in duplet for Selected node weights
+    selectedColorKnots : (number|Color3)[]
+    //0-1 number, then a color3 in duplet for skeleton color map
+    mapColorKnots : (number|Color3)[]
+}
+
+/*interface IBoneWeightShaderGradientCache{
+    selectedWeight: Nullable<number[]>|DynamicTexture
+    colorMap: Nullable<number[]>|DynamicTexture
+}*/
+
+export class BoneWeightShader{    
+    public name?: Nullable<string>    
+    public skeleton: Skeleton
+    public mesh: AbstractMesh
+    private _scene: Scene  
+    //private _renderingMode: number
+    //private _gradientCache: Nullable<number[]> 
+
+    constructor(
+        public options: Partial<IBoneWeightOptions> = {}
+    ){
+
+        this.options.renderingMode = options.renderingMode || BoneWeightShader.MODE_SELECTED_WEIGHT 
+        this.options.selectedColorKnots = options.selectedColorKnots || [0.25, Color3.Blue(), 0.5, Color3.Green(), 0.75, Color3.Yellow(), 1, Color3.Red()],
+        this.options.mapColorKnots = options.mapColorKnots || [0.25, Color3.Purple(), 0.5, (Color3.Red().add(Color3.Yellow())).scale(0.5), 0.75, Color3.Green(), 1, Color3.Teal()] 
+
+        this._scene = this.skeleton.getScene()
+        //this._renderingMode = this.options.renderingMode      
+    }
+
+    _prep(){

```

Comment:
This is not yet called right?

Reply:
No I was just getting it in place.

Reply:
the weight shader is not called by anything yet but will here in the next few commits.